spring-batch-core/src/main/java/org/springframework/batch/core/job/AbstractJob.java

```diff
@@ -60,6 +63,8 @@
 public abstract class AbstractJob implements Job, StepLocator, BeanNameAware,
 InitializingBean {
 
+	private static final String METRICS_PREFIX = "spring.batch.";
```

Comment:
@mminella What is the best place to define this constant? It is currently duplicated in 4 places.

Reply:
I'm struggling with an answer here.  The reason is that I can't think of anywhere else in the framework (short of logging) that we have a library like this that we have sprinkled through out the framework in this decentralized way.  Because of that, there isn't an obvious place to hang that constant (or the `SUCCESS`/`FAILURE` ones either for that matter).  I considered suggesting putting the logic we need into a `MetricsHandler` or something of the like that would encapsulate just our usage of micrometer (static util methods).  That would be a logical place to drop these constants as well.  However, you then get closer to the facade of a facade that we were trying to avoid with that approach (it's not the same thing, but it is similar)...

Reply:
Good idea. I moved all interactions with micrometer to a `BatchMetrics` class. This makes the code cleaner and is a good place to define metrics constants.